### ISSUE TITLE
Convert to pointer crash on Windows

### ISSUE THREAD

**Issue by CraftSpider** [aldelaro5/Dolphin-memory-engine#26]
On Windows 10, sometimes when I try to change an existing watch address to a pointer, the DME stops responding and then crashes. This seems to occur more frequently the longer the program has been open.

**Comment by aldelaro5** [comment]
I found out the cause and a way to reproduce this issue.

The problem is that the way I did the pointer offset is very weird: if you load the watch entry dialog to edit one when it is NOT a pointer, it will add an offset as a placeholder in case you decide to check that it is a pointer.

The issue is this is saved regardless of whether or not it was ACTUALLY a pointer so the UI assumes there's an offset to display when there isn't so it segfaults.

The correct way to implement it would have been to NOT touch the entry on the constructor, but only when checking or unchecking the checkbox.  This is pretty dumb, but I should be able to get with a fix soon.
